src/main/java/org/jboss/ws/tools/security/legacy/SecurityManagerUtils.java

```diff
@@ -0,0 +1,107 @@
+package org.jboss.ws.tools.security.legacy;
+
+import java.lang.reflect.Method;
+
+/**
+ * A helper class that caches and provides status about the SecurityManager API, which is deprecated starting from
+ * JDK17 (JEP 411), and removed on JDK 23+ (JEP 486).
+ * It also keeps the java.security.* deprecated APIs separate.
+ */
+public class SecurityManagerUtils {
```

Comment:
Similar here:
 * don't allow inheritance by default

I would recommend to:
 * make this class final
 * make constructor private